### PR TITLE
Publisher field and society field cannot have the same answer

### PR DESCRIPTION
**Pull request by amdomanska**
# Publisher field and society field cannot have the same answer

*Please don't delete any sections when completing this PR template; instead enter **N/A** for checkboxes or sections which are not applicable, unless otherwise stated below*

[See #3136](https://github.com/DOAJ/doajPM/issues/3163)

1. DifferentTo custom parsley validator should be case insensitive (eg. "CottageLabs Publishing" is the same as "cottagelabs publishing" - fixed
2. Messages from different_to validator in application_forms.py were not passed to the parsley validator - fixed
3. different_to validator added to `Publisher's name` and `Institution name` fields as required

## Categorisation

This PR...
- [ ] has scripts to run
- [ ] has migrations to run
- [ ] adds new infrastructure
- [ ] changes the CI pipeline
- [x] affects the public site
- [x] affects the editorial area
- [x] affects the publisher area
- [ ] affects the monitoring

## Basic PR Checklist

Instructions for developers:
* For each checklist item, if it is N/A to your PR check the N/A box
* For each item that you have done and confirmed for yourself, check Developer box (including if you have checked the N/A box)

Instructions for reviewers:
* For each checklist item that has been confirmed by the Developer, check the Reviewer box if you agree
* For multiple reviewers, feel free to add your own checkbox with your github username next to it if that helps with review tracking

### Code Style

- No deprecated methods are used
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer

- No magic strings/numbers - all strings are in `constants` or `messages` files
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer
  
- ES queries are wrapped in a Query object rather than inlined in the code
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer
  
- Where possible our common library functions have been used (e.g. dates manipulated via `dates`)
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer
  
- Cleaned up commented out code, etc
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer

### Testing

- Unit tests have been added/modified
  - [ ] N/A
  - [ ] Developer
  - [ ] Reviewer
  
- Functional tests have been added/modified
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer
  
- Code has been run manually in development, and functional tests followed locally
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer

### Documentation

- FeatureMap annotations have been added
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer
  
- Documentation updates - if needed - have been identified and prepared for inclusion into main documentation (e.g. added and highlighted/commented as appropriate to this PR)
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer
  
- Core model documentation has been added to if needed: https://docs.google.com/spreadsheets/d/1lun2S9vwGbyfy3WjIjgXBm05D-3wWDZ4bp8xiIYfImM/edit
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer

- Events and consumers documentation has been added if needed: https://docs.google.com/spreadsheets/d/1oIeG5vg-blm2MZCE-7YhwulUlSz6TOUeY8jAftdP9JE/edit
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer
  
- The docs for this branch have been generated and pushed to the doc site (see docs/README.md for details)
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer

### Release Readiness

- If needed, migration has been created and tested locally
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer

- Release sheet has been created, and completed as far as is possible https://docs.google.com/spreadsheets/d/1Bqx23J1MwXzjrmAygbqlU3YHxN1Wf7zkkRv14eTVLZQ/edit
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer

- There has been a recent merge up from `develop` (or other base branch).  List the dates of the merges up from develop below
  - 4/08/2023

## Testing

List the Functional Tests that must be run to confirm this feature

1. [New Application Form/Publisher's Form](https://doaj.github.io/doaj-docs/feature/3136_different_publisher_and_society/testbook/index.html#new_application_form/publishers_form)

## Deployment

N/A

### Configuration changes

N/A

### Scripts

N/A

### Migrations

N/A

### Monitoring

N/A

### New Infrastructure

N/A

### Continuous Integration

N/A

### CODE REVIEW

**Review by RK206 — APPROVED** [review]
Looks good

**Review by philipkcl — APPROVED** [review]

**Review by Steven-Eardley — APPROVED** [review]
Brilliant

File: portality/static/js/application_form.js [hunk 1]
```diff
@@ -844,8 +844,8 @@ window.Parsley.addValidator("optionalIf", {
 });
 
 window.Parsley.addValidator("differentTo", {
-    validateString : function(value, requirement) {
-      return (!value || ($("[name = " + requirement + "]")).val() !== value);
+    validateString : function(value, requirement, message) {
+      return (!value || ($("[name = " + requirement + "]")).val().toLowerCase() !== value.toLowerCase());
```
Comment (richard-jones) [inline portality/static/js/application_form.js:848]: This function takes the argument `message` but doesn't use it, is that required by the Parsley interface or something?

Comment (amdomanska) [reply portality/static/js/application_form.js:848]: No, that's overlook. Sorry!

File: portality/forms/application_forms.py [hunk 2]
```diff
@@ -547,10 +547,27 @@ class FieldDefinitions:
...
+                "widgets": [
+                    "click_to_copy",  # ~~^-> ClickToCopy:FormWidget~~
+                ]
+            },
+            "editor": {
+                "widgets": [
+                    "click_to_copy",  # ~~^-> ClickToCopy:FormWidget~~
+                ]
+            }
+        },
```
Comment (richard-jones) [inline portality/forms/application_forms.py:566]: THis overwrites the widgets on the base field, so it needs to include the `autocomplete` and any other widgets that you also want active on the admin form.

**Review by richard-jones — CHANGES_REQUESTED** [review]
Ensure context specific widget config doesn't overwrite widgets from the base field behaviour

**Pull request closed without merging**